### ISSUE TITLE
[REQUEST] Add Kill Aura 

### ISSUE THREAD

**Issue by Bavfalcon9** [Bavfalcon9/Mavoric#30]
**Description:** There is currently no solution for kill aura detection(s).

**Resolution:** There are multiple ways to go about this, so I will be thinking about them throughout the week.

**Comment by DarkWav** [comment]
Hello there, "retired" AntiCheat developer here. (just coding my AC for fun right now).

Correct me if I'm wrong, but the current implementation seems to still allow hitting targets the player is not actually facing, so if the KillAura doesn't even aim on the target and just hits targets behind the player the detection is useless. It just detects head snap, which many KillAura variants won't even perform.

A solution to this issue would be to actually measure the Angle difference between the players direction vector and a vector from the player towards the attacked entity.

In order to compare and generate a proper angle difference these vectors also need to be normalized.

Another thing to consider is that this check should only trigger when the two entities are at least ~1.5 blocks away from each other since mobile players can actually hit entities ~1.5 blocks behind themselves when looking straight to the ground by taping on the edges of their screen.

Wether to respect Y axis in the measurement or not is another topic that needs to be considered.
If you are interested, I could make a pull request with a possible implementation of some angle-based checking.

**Comment by Olybear9** [comment]
Hi, appreciate the comment. Theres multiple issues with kill aura and I've found a few solutions to them as I go, and ironically all the cheats on v1.0.0. First thing, angle checks are often false triggered, whether this is lag, or just irregular head movement from a genuine player, which is what I was trying to fix on my local version, I've found a fix which I will probably be pushing soon. Secondly, snap checks are purely experimental and probably won't be added to the final version, or at least to the degree its coded in right now, as simply spinning with high sensitivity triggers a snap. 

I will look into your suggestion seeing if theres more I can add to kill aura as I go, so I appreciate the comment.

Another thing to consider is that, Mavoric was designed to be light weight, so some of these checks are poor in order to comply with it. This goal has been removed in v2.0.0 with the implementation of task checks and a lot of other things to help improve accuracy. Like you, I'm not happy with any v1.0.x versions which is why im working on 2.0.0. But thanks again for the comment, and I will look into more implementation with your suggestion.
